lib/migrator.js

```diff
@@ -48,27 +68,32 @@ internals.executeMigration = async (migration, rollback) => {
     catch (err) {
 
         // Even if our symlinking fails, we need to manually cleanup our temporary directory because
-        // the unsafeCleanup: true option we pass to tempDirAsync (Tmp.dir) causes the tool to not delete the temp directory
-        // if it has entries, which it will if we fail partway through
-        tmpResults.cleanup();
+        // the unsafeCleanup: true option we pass to tempDir (Tmp.dir promisified) causes Tmp to not delete the temp directory if it has entries,
+        // which it will if we fail partway through
+        tmpCleanup();
+        throw err;
     }
 
-    // TODO Review / fix tmpAssets.tmpPath
-    const relMigrationsDir = Path.relative(process.cwd(), tmpResults.tmpPath);
+    const relMigrationsDir = Path.relative(process.cwd(), tmpPath);
     const latestOrRollback = rollback ? 'rollback' : 'latest';
 
     await knex.migrate[latestOrRollback]({ directory: relMigrationsDir });
 
-    // TODO does this work? What happens when a callback takes 2 non-error parameters?
-    tmpResults.cleanup();
+    tmpCleanup();
+
 };
 
 internals.readMigrationDir = async (migrationDir, symlinkConfig) => {
 
-    const migrationFiles = await internals.PROMISIFIED_UTILITIES.readDirAsync(migrationDir);
+    const migrationFiles = await Util.promisify(Fs.readdir)(migrationDir);
+
+    // We store our current migrationDir here and not in the above for...of loop in executeMigration because if assigned there, subsequent iterations of that loop would overwrite the copy of symlinkConfig that the current iteration of readMigrationDir is reading and using to create symlinks
+    // So, the created symlinks would use the files read from the current migrationDir, but then the migrationDir from the next iteration, resulting in ENOENT errors
+    symlinkConfig.migrationDir = migrationDir;
+
     const migrationSymlinks = [];
 
-    for (const file in migrationFiles) {
+    for (const file of migrationFiles) {
```

Comment:
Is this fixing a previous change you made, or is this from the original migration code?

Reply:
fixing a previous change. I think writing `in` was just a (subtly unfortunate 🎺 ) typo on my part that caused things to break.